common/script/aria.js

```diff
@@ -7,746 +20,395 @@
-            dRef.id = "desc-" + title;
-            dRef.setAttribute("role", "definition");
-            var heading = document.createElement("h4");
-            heading.appendChild(sp);
-            container.replaceChild(heading, item);
-            // add this item to the index
-            propList[title] = {
-                is: type,
-                title: title,
-                name: content,
-                desc: desc,
-                roles: [],
-            };
-            var abstract = container.querySelector(
-                "." + type + "-applicability"
-            );
-            if (
-                (abstract.textContent || abstract.innerText) ===
-                "All elements of the base markup"
-            ) {
-                globalSP.push({
-                    is: type,
-                    title: title,
-                    name: content,
-                    desc: desc,
-                    prohibited: false,
-                    deprecated: false,
-                });
-            } else if (
-                (abstract.textContent || abstract.innerText) ===
-                "All elements of the base markup except for some roles or elements that prohibit its use"
-            ) {
-                globalSP.push({
-                    is: type,
-                    title: title,
-                    name: content,
-                    desc: desc,
-                    prohibited: true,
-                    deprecated: false,
-                });
-            } else if (
-                (abstract.textContent || abstract.innerText) ===
-                "Use as a global deprecated in ARIA 1.2"
-            ) {
-                globalSP.push({
-                    is: type,
-                    title: title,
-                    name: content,
-                    desc: desc,
-                    prohibited: false,
-                    deprecated: true,
-                });
-            }
-            // the rdef is gone.  if we are in a div, convert that div to a section
-
-            if (container.nodeName.toLowerCase() == "div") {
-                // change the enclosing DIV to a section with notoc
-                var sec = document.createElement("section");
-                Array.prototype.slice
-                    .call(container.attributes)
-                    .forEach(function (attr) {
-                        sec.setAttribute(attr.name, attr.value);
-                    });
-                sec.classList.add("notoc");
-                var theContents = container.innerHTML;
-                sec.innerHTML = theContents;
-                container.parentNode.replaceChild(sec, container);
-            }
+/**
+ *
+ * @param {HTMLElement} container - parent of sdef or pdef or rdef
+ */
+const rewriteDefContainer = (container) => {
+    // if we are in a div, convert that div to a section
+    // TODO:
+    // a) seems to be always the case.
+    // b) Why don't we author the spec this way?
```

Comment:
Copied from https://github.com/w3c/aria-common/pull/106#discussion_r1462542533

> Seems like we should. Anything that is not generated should just be in the static html. It's odd because this container has `id=<attributename>` on it for the url fragment, which probably should be generated, since it is the same as the pdef/sdef